lib/ansible/modules/source_control/gitlab_project_variable.py

```diff
@@ -0,0 +1,262 @@
+    default: false
+    required: false
+    type: bool
+  vars:
+    description:
+      - A list of key value pairs.
+    default: {}
+    required: false
+    type: dict
+'''
+
+
+EXAMPLES = '''
+- name: Set or update some CI/CD variables
+  gitlab_project_variable:
+    api_url: https://gitlab.com
+    api_token: secret_access_token
+    project: markuman/dotfiles
+    purge: False
+    vars:
+      ACCESS_KEY_ID: abc123
+      SECRET_ACCESS_KEY: 321cba
+
+- name: Delete one variable
+  gitlab_project_variable:
+    api_url: https://gitlab.com
+    api_token: secret_access_token
+    project: markuman/dotfiles
+    state: absent
+    vars:
+      ACCESS_KEY_ID: abc123
+'''
+
+RETURN = '''
+msg:
+  description: Success or failure message
+  returned: failed
+  type: str
+  sample: "Success"
+
+project_variable:
+  description: Four lists of the variablenames which were added, updated, removed or exist.
+  returned: always
+  type: dict
+  contains:
+    added:
+      description: A list of variables which were created.
+      returned: always
+      type: list
+      sample: "['ACCESS_KEY_ID', 'SECRET_ACCESS_KEY']"
+    untouched:
+      description: A list of variables which exist.
+      returned: always
+      type: list
+      sample: "['ACCESS_KEY_ID', 'SECRET_ACCESS_KEY']"
+    removed:
+      description: A list of variables which were deleted.
+      returned: always
+      type: list
+      sample: "['ACCESS_KEY_ID', 'SECRET_ACCESS_KEY']"
+    updated:
+      description: A list of variables whose values were changed.
+      returned: always
+      type: list
+      sample: "['ACCESS_KEY_ID', 'SECRET_ACCESS_KEY']"
+'''
+
+import traceback
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+from ansible.module_utils._text import to_native
+from ansible.module_utils.api import basic_auth_argument_spec
+
+
+GITLAB_IMP_ERR = None
+try:
+    import gitlab
+    HAS_GITLAB_PACKAGE = True
+except Exception:
+    GITLAB_IMP_ERR = traceback.format_exc()
+    HAS_GITLAB_PACKAGE = False
+
+
+class gitlab_project_variables(object):
+
+    def __init__(self, module):
+        self.repo = gitlab.Gitlab(module.params['api_url'],
+                                  private_token=module.params['api_token'])
+        self.project = self.get_project(module.params['project'])
+        self._module = module
+
+    def auth(self):
+        self.repo.auth()
+
+    def get_project(self, project_name):
+        return self.repo.projects.get(project_name)
+
+    def list_all_project_variables(self):
+        return self.project.variables.list()
+
+    def create_variable(self, key, value):
+        if self._module.check_mode:
+            return
+        return self.project.variables.create({"key": key, "value": value})
+
+    def update_variable(self, var, value):
+        if var.value == value:
+            return False
+        if self._module.check_mode and var.value != value:
```

Comment:
```suggestion
        if self._module.check_mode:
```
You already checked for equality above.